x-pack/plugins/infra/public/pages/logs/shared/use_log_ml_job_id_formats_shim.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import createContainer from 'constate';
+import { useState, useEffect, useMemo, useCallback } from 'react';
+import { useLogViewContext } from '@kbn/logs-shared-plugin/public';
+import { IdFormatByJobType, JobType } from '../../../../common/http_api/log_analysis/v1/id_formats';
+import { useActiveKibanaSpace } from '../../../hooks/use_kibana_space';
+import { useTrackedPromise } from '../../../utils/use_tracked_promise';
+
+import { useKibanaContextForPlugin } from '../../../hooks/use_kibana';
+import { callGetLogAnalysisIdFormats } from './call_get_log_analysis_id_formats';
+
+const useLogMlJobIdFormatsShim = () => {
+  const [idFormats, setIdFormats] = useState<IdFormatByJobType | null>(null);
+
+  const { logViewReference } = useLogViewContext();
+  const { space } = useActiveKibanaSpace();
+  const { services } = useKibanaContextForPlugin();
+
+  const [getLogAnalysisIdFormatsRequest, getLogAnalysisIdFormats] = useTrackedPromise(
+    {
+      cancelPreviousOn: 'creation',
+      createPromise: async () => {
+        if (!space) {
+          return { data: null };
+        }
+
+        if (logViewReference.type === 'log-view-inline') {
+          throw new Error('Logs ML features only support persisted Log Views');
+        }
+
+        return await callGetLogAnalysisIdFormats(
+          {
+            logViewId: logViewReference.logViewId,
+            spaceId: space.id,
+          },
+          services.http.fetch
+        );
+      },
+      onResolve: ({ data }) => {
+        setIdFormats(data);
+      },
+    },
+    [logViewReference, space]
+  );
+
+  useEffect(() => {
+    getLogAnalysisIdFormats();
+  }, [getLogAnalysisIdFormats]);
+
+  const isLoadingLogAnalysisIdFormats = useMemo(
+    () => getLogAnalysisIdFormatsRequest.state === 'pending',
+    [getLogAnalysisIdFormatsRequest.state]
+  );
+
+  const hasFailedLoadingLogAnalysisIdFormats = useMemo(
+    () => getLogAnalysisIdFormatsRequest.state === 'rejected',
+    [getLogAnalysisIdFormatsRequest.state]
+  );
+
+  const migrateIdFormat = useCallback(
+    (jobType: JobType) => {
+      setIdFormats({
+        ...idFormats!,
+        [jobType]: 'hashed',
+      });
+    },
+    [idFormats]
+  );
```

Comment:
Would this work?

```suggestion
 const migrateIdFormat = useCallback((jobType: JobType) => {
    setIdFormats(
      (value) => ({ ...(value ? value : {}), [jobType]: 'hashed' } as IdFormatByJobType)
    );
  }, []);
```

Reply:
That wouldn't work like that because it only migrates one job type at a time and the type expects values for both job types to be present (which it should be if it has actually loaded the formats from the API).
But I'll do something similar which is to just keep setting it to null if it hasn't loaded it from the API yet which I think would be more correct.